set-web/src/main/java/eu/europeana/set/web/service/UserSetService.java

```diff
@@ -1,184 +1,202 @@
+	
+	/**
+	 * This method disables user set in database
+	 * @param existingUserSet
+	 * @return disabled user set
+	 */
+	public UserSet disableUserSet(UserSet existingUserSet);					 
+	
+	/**
+	 * This method updates user set pagination values. 
+	 * @param newUserSet
+	 * @return user set with updated pagination values
+	 */
+	public void updateUserSetPagination(UserSet newUserSet);		
+	
+	/**
+	 * This method returns UserSet object for given user set identifier.
+	 * @param
+	 * @return UserSet object
+	 */
+	public UserSet getUserSetById(String userSetId) throws UserSetNotFoundException; 
+		
+	/**
+	 * This method returns UserSet object for given user set identifier. Additionally
+	 * this method allows to define whether disabled user sets should be searched.
+	 * Use false if disabled user sets should be returned.
+	 * @param user set ID
+	 * @param true if disabled user sets should be checked (default true)
+	 * @return UserSet object
+	 */
+	public UserSet getUserSetById(String userSetId, boolean checkDisabled) throws UserSetNotFoundException; 
+		
+	/**
+	 * This method forms an identifier URL
+	 * @param id The sequential ID
+	 * @param base The base URL
+	 * @return identifier URL
+	 */
+	public String buildIdentifierUrl(String id, String base);
+	
+    /**
+     * This method computes pagination values for user set
+     * @param userSet
+     * @return enriched user set
+     */
+    public UserSet fillPagination(UserSet userSet);
+	
+    /**
+     * This method computes pagination values for user set without update of identifier
+     * @param userSet
+     * @return enriched user set
+     */
+    public UserSet updatePagination(UserSet userSet);
+	
+	/**
+	 * This methods converts user set object from JsonLd string format to a UserSet object
+	 * @param userSetJsonLdStr
+	 * @return a UserSet object
+	 * @throws HttpException
+	 */
+	public UserSet parseUserSetLd(String userSetJsonLdStr) throws HttpException;
+
+	/**
+	 * This method validates and processes the Set description for format and mandatory fields
+	 * if false responds with HTTP 400
+	 * @param webUserSet
+	 * @throws RequestBodyValidationException 
+	 */
+	public void validateWebUserSet(UserSet webUserSet) throws RequestBodyValidationException;
+	
+	/**
+	 * This method validates and processes the favorite set
+	 * @param webUserSet The new user set
+	 * @param existingUserSet The stored user set
+	 * @throws RequestBodyValidationException
+	 */
+	public void validateFavoriteUserSet(UserSet webUserSet, UserSet existingUserSet) throws RequestBodyValidationException, UserSetNotFoundException;
+
+	/**
+	 * This method checks if a user set with provided type and user already exists in database
+	 * @param type
+	 * @param creator
+	 * @return true if a user set object found in database
+	 * @throws UserSetNotFoundException
+	 */
+	public boolean isTypeAndCreatorExisting(String type, Agent creator) throws UserSetNotFoundException;
```

Comment:
is it required to include this method in the interface? if this is only used for validation, than not

Reply:
It is not required indeed. I removed it from interface.